code/modules/organs/external/_external.dm

```diff
@@ -365,44 +365,47 @@
 	return all_limbs
 
 /obj/item/organ/external/proc/is_dislocated()
-	if(dislocated > 0)
-		return 1
-	if(is_parent_dislocated())
-		return 1//if any parent is dislocated, we are considered dislocated as well
-	return 0
+	return (dislocated > 0) || is_parent_dislocated() //if any parent is dislocated, we are considered dislocated as well
 
 /obj/item/organ/external/proc/is_parent_dislocated()
 	var/obj/item/organ/external/O = parent
-	while(O && O.dislocated != -1)
-		if(O.dislocated == 1)
-			return 1
+	while(O && (O.limb_flags & ORGAN_FLAG_CAN_DISLOCATE))
+		if(O.dislocated > 0)
+			return TRUE
 		O = O.parent
-	return 0
+	return FALSE
 
 /obj/item/organ/external/proc/update_internal_organs_cost()
 	internal_organs_size = 0
 	for(var/obj/item/organ/internal/org in internal_organs)
 		internal_organs_size += org.get_storage_cost()
 
 /obj/item/organ/external/proc/dislocate()
-	if(dislocated == -1)
+	if(owner && owner.status_flags & GODMODE)
+		return
+	if(!(limb_flags & ORGAN_FLAG_CAN_DISLOCATE))
 		return
 
-	dislocated = 1
+	dislocated = TRUE
 	if(owner)
+		if(owner.can_feel_pain(src))
```

Comment:
Believe `add_pain` already takes can_feel_pain into account and will handle stuff like limb-specific numbness better than `owner.can_feel_pain()`. Checking code, calling owner.can_feel_pain with src is strictly redundant in view of it just calling src.can_feel_pain() anyway.

Reply:
By limb specific numbness what do you mean? Like does it adds extra effects?
Also, the main reason for the check is for the owner.apply_effect(). Putting add_pain() in there alongside it just means it'll do the check a single time instead of twice if I've had placed add_pain() outside.

Reply:
Conceptually, if we later want to add some kind of per-limb pain response, pain handling, etc. then this change wouldn't be ideal due to prioritizing mob-level responses over limb-level. The main reason I think you should use limb-level pain procs is that as discussed on Discord the mob level procs just wrap the limb level ones anyway.

Reply:
Yeah, I'll change it to the limb level proc in a bit.